### PR TITLE
add `axis` to `cupy.logspace`

### PR DESCRIPTION
**Pull request by IncubatorShokuhou**
Add `axis` to `cupy.logspace`

### CODE REVIEW

**Comment by emcastillo** [comment]
Thanks for the change! Would you mind adding some tests?

**Comment by IncubatorShokuhou** [comment]
> Thanks for the change! Would you mind adding some tests?

A test added, but I am not sure if I added the test correctly. Would you please review it?

File: tests/cupy_tests/creation_tests/test_ranges.py [hunk 1]
```diff
@@ -284,6 +284,14 @@ def test_logspace_neg_num(self):
     def test_logspace_base(self, xp, dtype):
         return xp.logspace(0, 2, 5, base=2.0, dtype=dtype)
```
Comment (emcastillo) [inline tests/cupy_tests/creation_tests/test_ranges.py:286]:
```suggestion


```

Comment (emcastillo) [reply tests/cupy_tests/creation_tests/test_ranges.py:286]: 2 lines due to format

Comment (IncubatorShokuhou) [reply tests/cupy_tests/creation_tests/test_ranges.py:286]: Resolved.

File: tests/cupy_tests/creation_tests/test_ranges.py [hunk 2]
```diff
@@ -285,6 +285,15 @@ def test_logspace_base(self, xp, dtype):
...
 
+    @testing.with_requires('numpy>=1.16')
+    @testing.for_all_dtypes_combination(names=('dtype_range', 'dtype_out'),
+                                        no_bool=True, no_complex=True)
+    @testing.numpy_cupy_array_equal()
+    def test_logspace_array_start_stop_axis1(self, xp, dtype_range, dtype_out):
+        start = xp.array([0, 2], dtype=dtype_range)
+        stop = xp.array([2, 0], dtype=dtype_range)
+        return xp.logspace(start, stop, num=5, dtype=dtype_out, axis=1)
+
```
Comment (emcastillo) [inline tests/cupy_tests/creation_tests/test_ranges.py:296]: Sorry, here we need two lines too

**Comment by emcastillo** [comment]
I think you may need to change the tolerances in the tests to pass

**Comment by IncubatorShokuhou** [comment]
> I think you may need to change the tolerances in the tests to pass

It seems that there are  inconsistencies  between the result of `np.logspace(start, stop, num=5,axis=1)` and ``np.logspace(start, stop, num=5,axis=1)`:
```
================================== FAILURES ===================================
_______________ TestRanges.test_logspace_array_start_stop_axis1 _______________
C:\Development\Python\Python310\lib\site-packages\cupy\testing\_loops.py:1078: in test_func
    impl(*args, **kw_copy)
C:\Development\Python\Python310\lib\site-packages\cupy\testing\_loops.py:353: in test_func
    check_func(cupy_r, numpy_r)
C:\Development\Python\Python310\lib\site-packages\cupy\testing\_loops.py:658: in check_func
    _array.assert_array_equal(x, y, err_msg, verbose, strides_check)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x = array([[ 1.       ,  3.1622777, 10.       , 31.622776 , 99.99999  ],
       [99.99999  , 31.622776 , 10.       ,  3.1622777,  1.       ]],
      dtype=float32)
y = array([[  1.       ,   3.1622777,  10.       ,  31.622776 , 100.       ],
       [100.       ,  31.622776 ,  10.       ,   3.1622777,   1.       ]],
      dtype=float32)
```
What should I do ?  Should I change the test case, or what else should I do?

**Comment by emcastillo** [comment]
in `@testing.numpy_cupy_allclose(rtol=1e-6)` you can change `atol` and `rtol`

**Comment by IncubatorShokuhou** [comment]
> in `@testing.numpy_cupy_allclose(rtol=1e-6)` you can change `atol` and `rtol`
OK. `rtol=1e-6`  has been set.

**Comment by emcastillo** [comment]
Can you disable contiguity check in the test? thanks!!

**Comment by IncubatorShokuhou** [comment]
> Can you disable contiguity check in the test? thanks!!

How to disable it?

**Comment by emcastillo** [comment]
`@testing.numpy_cupy_allclose(contiguous_check=False)`

**Comment by IncubatorShokuhou** [comment]
> disable contiguity check

Done.

**Comment by emcastillo** [comment]
just rebased on top of master and tried to skip the test for CUDA 10.2

**Pull request merged**